MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -14041,9 +14042,38 @@ void ApplicationWindow::updateRecentFilesList(QString fname) {
     recentFiles.pop_back();
 
   recentFilesMenu->clear();
-  for (int i = 0; i < (int)recentFiles.size(); i++)
-    recentFilesMenu->insertItem("&" + QString::number(i + 1) + " " +
-                                recentFiles[i]);
+  for (int i = 0; i < (int)recentFiles.size(); i++) {
+    QMenuItem *mi = new QMenuItem;
+
+    try {
+      Mantid::API::MultipleFileProperty mfp("tester");
+      mfp.setValue(recentFiles[i].toStdString());
+      const std::vector<std::string> files =
+          Mantid::API::MultipleFileProperty::flattenFileNames(mfp());
+      if (files.size() == 1) {
+        std::ostringstream ostr;
+        ostr << "&" << QString::number(i + 1).toStdString() << " " << files[0];
+        mi->setText(QString::fromStdString(ostr.str()));
+      } else if (files.size() > 1) {
+        std::ostringstream ostr;
+        ostr << "&" << QString::number(i + 1).toStdString() << " " << files[0]
```

Comment:
Since paths that no longer exist are skipped, `i + 1` is not always the correct file number anymore. If the third and fourth files in your history have been deleted, for example, the menu will count 1, 2, 5, 6,...